src/main/java/com/a2/backend/service/impl/ProjectServiceImpl.java

```diff
@@ -93,4 +93,29 @@ public Project getProjectDetails(UUID projectID) {
                                 new ProjectNotFoundException(
                                         String.format("No project found for id: %s", projectID)));
     }
+
+    @Override
+    public List<Project> getProjectsByTitleSearch(String pattern) {
+        List<Project> projectsStartingWithPattern =
+                projectRepository.findByTitleStartsWithIgnoreCaseOrderByTitleAsc(pattern);
+        List<Project> projectsContainingPattern =
+                projectRepository.findByTitleContainingIgnoreCaseOrderByTitleAsc(pattern);
+
+        for (int i = 0; i < projectsContainingPattern.size(); i++) {
+            for (int j = 0; j < projectsStartingWithPattern.size(); j++) {
+                if (projectsContainingPattern
+                        .get(i)
+                        .getTitle()
+                        .equals(projectsStartingWithPattern.get(j).getTitle())) {
+                    projectsContainingPattern.remove(i);
+                    i--;
```

Comment:
no entiendo por que el i--;

Reply:
por un tema que estoy eliminando elementos adentro del for, si no hago eso el for se saltea una posicion

Reply:
aaa ok, pero no te tira un infinite loop ? y si no hay projectos con ese pattern que pasa?